code/src/nuvla/connector/kubernetes.py

```diff
@@ -596,6 +614,385 @@ def reboot(self):
+        helm_repository = "nuvlaedge/nuvlaedge"
+        peripherals = self.get_helm_peripherals(modules)
+        env_vars = self.get_env_vars_string(install_params_from_payload)
+        working_dir = self.get_working_dir(install_params_from_payload)
+        job_lite_tag="issue-112-update-ne"
+        mandatory_args = \
+            " --set HOME=%s --set NUVLAEDGE_UUID=nuvlabox/%s \
+            --set kubernetesNode=$THE_HOST_NODE_NAME \
+            --set NUVLAEDGE_JOB_ENGINE_LITE_IMAGE=nuvladev/job-lite:%s \
+            --set vpnClient=true"\
+            %(f'{working_dir}',f'{project_uuid}',f'{job_lite_tag}',)
+        helm_namespace = " -n default"
+        helm_update_cmd = \
+            "helm upgrade %s %s %s --version %s %s %s %s"\
+                %(f'{project_name}',f'{helm_repository}', \
+                f'{helm_namespace}', \
+                f'{target_release}', \
+                f'{mandatory_args}', \
+                f'{peripherals}', \
+                f'{env_vars}',)
+
+        log.info(f"Helm upgrade command: \n {helm_update_cmd}")
+
+        return helm_update_cmd
+
+    def get_working_dir(self, install_params_from_payload):
+        """
+        Parse the payload and return a formatted string of helm environment variables
+        """
+        working_dir="/root"
+
+        if install_params_from_payload['working-dir']:
+            return install_params_from_payload['working-dir']
+        # do we test that the working dir exists?
+
+        return working_dir
+
+    def get_env_vars_string(self, install_params_from_payload):
+        """
+        Parse the payload and return a formatted string of helm environment variables
+        """
+
+        new_vars_string = ''
+        env_pair_pattern = r"^\w+=\w+$"
+
+        envs = install_params_from_payload['environment']
+        for env_pair in envs:
+            log.debug(f"Environment pair: {env_pair}")
+            env_pair_mod = "".join(env_pair.split())
+            re_result = re.match(env_pair_pattern,env_pair_mod)
+            log.debug(f"Matching result: {re_result}")
+            if re_result:
+                new_vars_string = new_vars_string + " --set " + env_pair_mod
+                log.debug(f"Current env var string: \n{new_vars_string}")
+
+        log.info(f"Environment list arguments: \n {new_vars_string}")
+
+        return new_vars_string
+
+    def create_job_name(self, the_job_name, k=5):
+        """
+        Create the job name with random string attached
+        """
+        new_job_name = the_job_name + "-" + ''.join(random.choices(string.digits, k=k))
+
+        return new_job_name
+
+    def check_target_release(self, helm_log_result, target_release, current_version):
+        """
+        Check the status of the target release
+        """
+        if target_release not in helm_log_result.stdout:
+            result = \
+            "Updating chart version change from %s to %s"\
+            %(f'{current_version}',f'{target_release}')
```

Comment:
There is multiples usage of "% formatting" which mix f-strings like this one.

With "% formatting" it could be written like the following:
```python
x  = 'A: % - B: %' % (a, b)
```
No need to do:
```python
x  = 'A: % - B: %' % (f'{a}', f'{b}')
```

When possible, I would try to only use f-strings:
```python
x  = f'A: {a} - B: {b}'
```

Reply:
Will parse through and pick up these...

Reply:
Picked them out and changed.